### PR TITLE
(fix) DlgTrackInfo/~Multi: use configured notation for key display

### PR DESCRIPTION
**Pull request by ronso0**
Previously, Track Properties did show the key text stored in the track, not the corresponding key of the notation selected in Key preferences. Furthermore, when selecting such a key literal text of  a different notation in DlgTrackInfoMulti, that string was always converted to 'Traditional' notation (select 5A, get Cm).

This fixes Track Properties and uses the new helper function in Key preferences.

### CODE REVIEW

File: src/library/dlgtrackinfo.cpp [hunk 1]
```diff
@@ -744,7 +748,10 @@ mixxx::UpdateResult DlgTrackInfo::updateKeyText() {
 }
 
 void DlgTrackInfo::displayKeyText() {
-    const QString keyText = m_trackRecord.getMetadata().getTrackInfo().getKeyText();
+    const QString keyText =
+            KeyUtils::tryConvertKeyString(
+                    m_trackRecord.getMetadata().getTrackInfo().getKeyText(),
+                    m_notation);
```
Comment (daschuer) [inline src/library/dlgtrackinfo.cpp:754]: The issue is probably caused of hard to used code. We have here a hint:
https://github.com/mixxxdj/mixxx/blob/9817e5eab23ce907f8e49bfdb0050fef9549257d/src/track/track.cpp#L1453
but not here: 
https://github.com/mixxxdj/mixxx/blob/24a84b9d64607bb2b2cdcfc270bf9cb885074f3a/src/track/trackinfo.h#L35

The later seems to ba a container fr the key text stored in track metadata. Do we want to rename (again) it or at least add a comment? 

I think here we want exactly what found in the metadata, right?  
https://github.com/mixxxdj/mixxx/blob/9817e5eab23ce907f8e49bfdb0050fef9549257d/src/library/browse/browsethread.cpp#L239

Comment (daschuer) [inline src/library/dlgtrackinfo.cpp:754]: I think it is better to use the same string the track object will return for display purpose: 
```suggestion
    const QString keyText = KeyUtils::keyToString(m_record.getKeys().getGlobalKey());
```
this should work without `m_notation` which can be probably be removed, (not tested)

File: src/track/keyutils.cpp [hunk 2]
```diff
@@ -538,6 +544,24 @@ KeyUtils::KeyNotation KeyUtils::keyNotationFromNumericValue(double value) {
     return static_cast<KeyNotation>(value_floored);
 }
 
+KeyUtils::KeyNotation KeyUtils::keyNotationFromString(const QString& notationName) {
```
Comment (daschuer) [inline src/track/keyutils.cpp:547]: Please keep this nice refactoring.

**Review by daschuer — CHANGES_REQUESTED** [review]
I can confirm the issue. The fix can be improved.

**Comment by ronso0** [comment]
You're right, that code ishard to use : P
I overlooked `KeyUtils::setNotation()` which makes `keyToString()` use the selected notation.

Here are my observations:
Throwing Snow - Iridis
| | |
|---|---|
|Tracks  | 1A |
Deck |    1A
Track Properties dialog |     G#m
-> Re-import from file |1A
| | | 
database |  G#m
file metadat|     G#m

**Comment by ronso0** [comment]
Yes, it does work without `m_notation`

**Comment by ronso0** [comment]
I have no energy to look into the [inconsistency described above](https://github.com/mixxxdj/mixxx/pull/14011#issuecomment-2542653733), though we should fix that IMO.

(btw I don' consider this here a blocker, it's probably been like this for quite some time, 2.4 and earlier I think)

**Review by daschuer — APPROVED** [review]
LGTM, thank you.

**Comment by ronso0** [comment]
Thanks @daschuer for your hints!

**Pull request merged**